CS483/CS483/Kartaclysm/StateMachine/Gameplay/StateRacing.cpp

```diff
@@ -89,13 +91,18 @@ void Kartaclysm::StateRacing::Enter(const std::map<std::string, std::string>& p_
 	m_pGameObjectManager->RegisterComponentFactory("GOC_KartController", ComponentKartController::CreateComponent);
 	m_pGameObjectManager->RegisterComponentFactory("GOC_AIDriver", ComponentAIDriver::CreateComponent);
 	m_pGameObjectManager->RegisterComponentFactory("GOC_Racer", ComponentRacer::CreateComponent);
+
+	m_pGameObjectManager->RegisterComponentFactory("GOC_Effect", ComponentEffect::CreateComponent);
 	
 	// Store passed context parameters and begin race
 	m_mContextParams = p_mContextParameters;
 	SendRaceInfoEvent();
 	BeginRace();
 
 	HeatStroke::AudioPlayer::Instance()->PlaySoundEffect("Assets/Sounds/engine.wav", true);
+
+	// I have to do this horrible spaghetti crap because game object manager isn't a SINGLETON
+	HeatStroke::CollisionManager::Instance()->m_pGameObjectManager = m_pGameObjectManager;
```

Comment:
At the very least, this should be a SetGameObjectManager() method. The variable should be private in the class with getters/setters, not just public. There's a proper way to handle spaghetti code that is less spaghetti